src/use-swr.ts

```diff
@@ -444,7 +450,12 @@ export const useSWRHandler = <Data = any, Error = any>(
       // - `force` is true, which is the start of polling
       // - or `timer` is not 0, which means the effect wasn't canceled
       if (refreshInterval && timer !== -1) {
-        timer = setTimeout(execute, refreshInterval)
+        // Use the passed interval
+        // ...or invoke the function with the updated data to get the interval
+        const interval = isFunction(refreshInterval)
+          ? refreshInterval(data)
+          : refreshInterval
+        timer = setTimeout(execute, interval)
```

Comment:
If `interval` is `0` we don't want to continue polling:

```suggestion
        if (interval) {
          timer = setTimeout(execute, interval)
        }
```

Reply:
Ah that's right. I should move the logic outside of the current if to prevent the double check.

Reply:
Yep that would be great!